components/camel-http/src/test/java/org/apache/camel/component/http/HttpProducerCustomHeaderTest.java

```diff
@@ -0,0 +1,105 @@
+package org.apache.camel.component.http;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.component.http.handler.HeaderValidationHandler;
+import org.apache.http.impl.bootstrap.HttpServer;
+import org.apache.http.impl.bootstrap.ServerBootstrap;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.http.HttpHeaders.HOST;
+
+public class HttpProducerCustomHeaderTest extends BaseHttpTest {
+
+    private static final String CUSTOM_HOST = "test";
+
+    private HttpServer localServer;
+
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        Map<String, String> expectedHeaders = new HashMap<>();
+        expectedHeaders.put(HOST,
+                            CUSTOM_HOST);
+
+        localServer = ServerBootstrap.bootstrap().
+                setHttpProcessor(getBasicHttpProcessor()).
+                setConnectionReuseStrategy(getConnectionReuseStrategy()).
+                setResponseFactory(getHttpResponseFactory()).
+                setExpectationVerifier(getHttpExpectationVerifier()).
+                setSslContext(getSSLContext()).
+                registerHandler("*",
+                                new HeaderValidationHandler("GET",
+                                                            "customHostHeader=test",
+                                                            null,
+                                                            getExpectedContent(),
+                                                            expectedHeaders))
+                .registerHandler("*",
+                                 new HeaderValidationHandler("GET",
+                                                             null,
+                                                             null,
+                                                             getExpectedContent(),
+                                                             null))
+                .create();
+
+        localServer.start();
+
+        super.setUp();
+    }
+
+    @After
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+
+        if (localServer != null) {
+            localServer.stop();
+        }
+    }
+
+    @Test
+    public void testHttpProducer_GivenCustomHostHeaderQuery_SetCustomHost() throws Exception {
```

Comment:
Does the names of the test have to contain `_` ?

Reply:
Of course it doesn't, i find it more readable this way.

Reply:
I would prefer if we don't use the underscore